src/ripple/app/ledger/tests/Ledger_test.cpp

```diff
@@ -0,0 +1,149 @@
+
+namespace ripple {
+namespace test {
+
+class Ledger_test : public beast::unit_test::suite
+{
+    void test_genesisLedger (bool sign)
+    {
+        std::uint64_t const xrp = std::mega::num;
+
+        auto master = createAccount ("masterpassphrase");
+
+        Ledger::pointer LCL = createGenesisLedger(100000*xrp, master);
+
+        Ledger::pointer ledger = std::make_shared<Ledger>(false, *LCL);
+
+        // User accounts
+        auto gw1 = createAccount ("gw1");
+        expect (gw1.pk != master.pk, "gw1.pk != master.pk");
+        expect (gw1.sk != master.sk, "gw1.sk != master.sk");
+        auto gw2 = createAccount ("gw2");
+        auto gw3 = createAccount ("gw3");
+        auto alice = createAccount ("alice");
+        auto mark = createAccount ("mark");
+
+        // Fund gw1, gw2, gw3, alice, mark from master
+        makeAndApplyPayment(master, gw1, 5000 * xrp, ledger, sign);
+        makeAndApplyPayment(master, gw2, 4000 * xrp, ledger, sign);
+        makeAndApplyPayment(master, gw3, 3000 * xrp, ledger, sign);
+        makeAndApplyPayment(master, alice, 2000 * xrp, ledger, sign);
+        makeAndApplyPayment(master, mark, 1000 * xrp, ledger, sign);
+
+        LCL = close_and_advance(ledger, LCL);
+        ledger = std::make_shared<Ledger>(false, *LCL);
+
+        // alice trusts FOO/gw1
+        makeTrustSet (alice, gw1, "FOO", 1, ledger, sign);
+
+        // mark trusts FOO/gw2
+        makeTrustSet (mark, gw2, "FOO", 1, ledger, sign);
+
+        // mark trusts FOO/gw3
+        makeTrustSet (mark, gw3, "FOO", 1, ledger, sign);
+
+        // gw2 pays mark with FOO
+        makeAndApplyPayment(gw2, mark, "FOO", ".1", ledger, sign);
+
+        // gw3 pays mark with FOO
+        makeAndApplyPayment(gw3, mark, "FOO", ".2", ledger, sign);
+
+        // gw1 pays alice with FOO
+        makeAndApplyPayment(gw1, alice, "FOO", ".3", ledger, sign);
+
+        LCL = close_and_advance(ledger, LCL);
+        ledger = std::make_shared<Ledger>(false, *LCL);
+
+        createOffer (mark, Amount (1, "FOO", gw1), Amount (1, "FOO", gw2), ledger, sign);
+        createOffer (mark, Amount (1, "FOO", gw2), Amount (1, "FOO", gw3), ledger, sign);
+        cancelOffer (mark, ledger, sign);
+        freezeAccount (alice, ledger, sign);
+
+        LCL = close_and_advance(ledger, LCL);
+        ledger = std::make_shared<Ledger>(false, *LCL);
+
+        makeAndApplyPayment(alice, mark, 1 * xrp, ledger, sign);
+
+        LCL = close_and_advance(ledger, LCL);
+        ledger = std::make_shared<Ledger>(false, *LCL);
+
+        pass ();
+    }
+
+    void test_unsigned_fails ()
+    {
+        std::uint64_t const xrp = std::mega::num;
+
+        auto master = createAccount ("masterpassphrase");
+
+        Ledger::pointer LCL = createGenesisLedger (100000 * xrp, master);
+
+        Ledger::pointer ledger = std::make_shared<Ledger> (false, *LCL);
+
+        auto gw1 = createAccount ("gw1");
+
+        auto tx = makePayment(master, gw1, 5000 * xrp, false);
```

Comment:
Not a correction, just an idea.  Since you are thinking about names.  `makePayment` returns an `STTx`, right?  You could call it `getPaymentTx`.  However the existing names don't bother me at all,